daffodil-core/src/main/scala/org/apache/daffodil/runtime1/ElementBaseRuntime1Mixin.scala

```diff
@@ -90,7 +93,7 @@ trait ElementBaseRuntime1Mixin { self: ElementBase =>
     // So for these cases we do not want to capture value length, since
     // they are handled by the parsers as necessary
     val capturedByParsers =
-      (isSimpleType && impliedRepresentation == Representation.Text) ||
+      (isSimpleType && (impliedRepresentation == Representation.Text || lengthKind == LengthKind.Delimited)) ||
```

Comment:
Is there no analogous flaw/situation for unparsers, or are they doing this stuff right?

Reply:
I think the complexity of unparsing actually makes it so we can completely rely on the CaptureValueLength unparsers--value unparsers don't actually have to do anything special.

For example, when parsing text, the string value parser handles both parsing the string AND removing padding, so the parser must also be responsible for calculating the value length that excludes that padding that it removed. So we can't use the special CaptureValueLength parsers in these cases.

But with unparsing, we have separate unparsers that unparse the value and unparse the padding, so we can wrap the CaptureValueLength unparsers around just value unparser. And this complexity of separate parsers is needed for all the suspension stuff, so that complex allows capture value length to be easy.

If we split parsing into separate parsers like we do for unparsing, we could probably get rid of this "captured by parsers" logic, and potentially get rid of a handful of potential bugs. Capturing value length in two different ways is just asking for issues like this. That might be a pretty sizable refactoring though.